docs/rfcs/0006-site-wide-notification.md

```diff
@@ -0,0 +1,48 @@
+## Site Wide Notifications
+
+#### Abstract
+
+As a member of the marketing team, I'd like to be able to post a message that will appear on all pages of the site, for example to advertise a promotion.
+
+#### Requirements
+
+- Brief description of the plan for adding, editing and removing the site-wide notification (see options below)
+- CMS user can add, edit and remove banner message through Wagtail. Or, if that is difficult, admin user can add, edit and remove banner message through the django admin
+- User can dismiss the banner by clicking on the X. It should stay hidden for the rest of the session
+
+##### Designs and Mockups
+- See, for example, [InVision](https://impactbnd.invisionapp.com/share/4TQEESTEYMZ#/screens/345573039)
+- This element should also be in the [course PSD](https://drive.google.com/file/d/10uAQ3emiF3ufd5eigYow0kxwA_51WiVD/view?usp=sharing)
+
+#### Architecture Changes
+
+##### Backend changes
+
+- Add `NotificationPage` in wagtail with following field
+    - notification
+
+- Add `UserNotification` model to save user's viewed state against notification.
+    - notification_id
+    - user_id
+
+- Add contextprocessor to pass notification to templates. 
+    - Filter all live notifications from `NotificationPage`
+    - Filter notifications which are not viewed by user from `UserNotification`,
+    - If notification(s) are available, show them to users
+
+- Add view `MarkNotificationViewed` to add entry in `UserNotification`, once user clicked on `close` button on notification.
+
+###### Open Questions:
+   - Do we want to show these notifications to anonymous users? In that case saving state in model won't work.
```

Comment:
Yes, we will absolutely need to display these notifications for anonymous users -- the use case is marketing messages.

Reply:
Since we need to support anonymous users, I would just store the dismissal state a cookie and not worry about the `UserNotification` data model.

Reply:
@rhysyngsun something like this https://stackoverflow.com/questions/38168728/advertisement-banner-show-only-once-in-browser-session ?

Reply:
yes, exactly

Reply:
@rhysyngsun I am using `sessionStorage` for saving state. Does this makes sense.
```
The sessionStorage property allows you to access a session Storage object. sessionStorage is similar to Window.localStorage, 
the only difference is while data stored in localStorage has no expiration set, data stored in sessionStorage gets cleared when the page session ends. 
A page session lasts for as long as the browser is open and survives over page reloads and restores.
```